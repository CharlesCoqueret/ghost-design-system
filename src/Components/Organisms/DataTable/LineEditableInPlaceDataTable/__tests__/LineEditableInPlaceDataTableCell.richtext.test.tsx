import React, { ReactElement } from 'react';
import { render } from '@testing-library/react';

describe('LineEditableInPlaceDataTableCell richtext case', () => {
  it('renders with richtext', () => {
    let onChangeCallback: (newValue: string) => void = () => {
      return;
    };
    jest.mock('../../../../Molecules/RichTextField', () => ({
      __esModule: true,
      RichTextField: (props: { onChange: typeof onChangeCallback }): ReactElement => {
        onChangeCallback = props.onChange;
        return <div>{JSON.stringify(props)}</div>;
      },
    }));

    const LineEditableInPlaceDataTableCell = require('../LineEditableInPlaceDataTableCell').default;
    const { ColumnType } = require('../../Common');

    const handleUpdateDataChangeMock = jest.fn();

    const { container } = render(
      <table>
        <tbody>
          <tr>
            <LineEditableInPlaceDataTableCell<{ richtext: string }>
              column={{
                dataIndex: 'richtext',
                editable: true,
                title: 'Richtext',
                type: ColumnType.RICHTEXT,
              }}
              extra={{
                editedRowIndex: 0,
              }}
              handleUpdateDataChange={handleUpdateDataChangeMock}
              row={{ richtext: 'richtext' }}
              rowIndex={0}
            />
          </tr>
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();

    if (onChangeCallback) {
      onChangeCallback('test');
    }

    expect(handleUpdateDataChangeMock).toBeCalledTimes(1);
    expect(handleUpdateDataChangeMock).toBeCalledWith(0, 'richtext', 'test');
  });
});
