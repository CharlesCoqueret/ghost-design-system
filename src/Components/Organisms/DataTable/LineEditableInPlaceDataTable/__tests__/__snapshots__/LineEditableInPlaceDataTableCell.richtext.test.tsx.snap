// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineEditableInPlaceDataTableCell richtext case LineEditableInPlaceDataTableCell renders with richtext 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class=""
        >
          <div>
            {"inputClassName":"gds-typography","inputValue":"richtext","name":"richtext","readOnly":false}
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
